pkg/controller/migmigration/task.go

```diff
@@ -520,6 +521,18 @@ func (t *Task) Run() error {
 		if restore == nil {
 			return errors.New("Restore not found")
 		}
+		report, err := t.ensureDestinationStagePodsStarted()
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+		if report.failed {
+			t.fail(DestinationStagePodsFailed, report.reasons)
+			break
+		}
+		if !report.started {
```

Comment:
@alaypatel07 `!report.started` looks like a typo. I think it would be sane to just remove this check alltogether. Adding a Requeue here doesn't make sense since it will always go forward to `hasRestoreCompleted()` call

Reply:
Yeah I'm a little confused about this as well. Is this because we already checked if it failed? So `!report.Started` means that its completed?

Reply:
@pranavgaikwad yes, since the report.started and report.failed are complementary, I think we can just check the failed and remove this check. However, on the failed, I will requeue so that this gets tried the next time

Reply:
`report.failed == !report.started == true` I'll collapse the two conditions and make it into one, and pull out the log entry

Reply:
fixed